packages/rule-schema-to-typescript-types/src/index.ts

```diff
@@ -69,9 +69,7 @@ function compileSchema(
 
   const refMap = new Map<string, string>();
   // we only support defs at the top level for simplicity
-  const defs = (schema.$defs ?? schema.definitions) as
-    | Record<string, JSONSchema4>
```

Comment:
removed this one since `schema.$defs` and `schema.definitions` are both `Record<string, JSONSchema4> | undefined;`

![image](https://github.com/typescript-eslint/typescript-eslint/assets/44593294/44e4370d-1490-47f1-aa61-566171ad593e)

Reply:
Nice!